src/Common/CurrentMetrics.cpp

```diff
@@ -82,6 +82,8 @@
     M(PartsInMemory, "In-memory parts.") \
     M(MMappedFiles, "Total number of mmapped files.") \
     M(MMappedFileBytes, "Sum size of mmapped file regions.") \
+    M(MMappedAllocs, "Total number of mmapped allocations") \
+    M(MMappedAllocBytes, "Sum bytes of mmapped allocations") \
```

Comment:
They can be ProfileEvents.

Reply:
not exactly. It is gauge, not counter.

Reply:
You would remove `sub` in that case making it more similar to the `IOBufferAllocs`|`IOBufferAllocBytes`.
But I see we already have `MMappedFiles` here so it would be inconsistent with it in that case...

Reply:
Just like `MMappedFiles`, `MMappedAllocs` records how many ongoing allocations using mmap, which is important for system, because too many mmap allocations may exceeds the limitation `vm.max_map_count`.  https://github.com/ClickHouse/ClickHouse/issues/37001

Reply:
makes sense, thanks for clarifying
can you fix the tests please?